de.fhdo.lemma.eclipse.ui/src/de/fhdo/lemma/eclipse/ui/specify_url_dialog/SpecifyUrlDialog.xtend

```diff
@@ -0,0 +1,274 @@
+package de.fhdo.lemma.eclipse.ui.specify_url_dialog
+
+import org.eclipse.swt.widgets.Shell
+import org.eclipse.jface.dialogs.TitleAreaDialog
+import org.eclipse.jface.dialogs.MessageDialog
+import org.eclipse.swt.widgets.Composite
+import org.eclipse.jface.dialogs.IDialogConstants
+import org.eclipse.swt.graphics.Point
+import org.eclipse.swt.layout.GridLayout
+import org.eclipse.jface.dialogs.IMessageProvider
+import org.eclipse.swt.SWT
+import org.eclipse.swt.layout.GridData
+import org.eclipse.swt.widgets.Label
+import org.eclipse.swt.widgets.Text
+import org.eclipse.swt.widgets.Button
+import org.eclipse.swt.widgets.DirectoryDialog
+import org.eclipse.swt.events.SelectionListener
+import java.net.URL
+import java.net.MalformedURLException
+import org.eclipse.core.resources.ResourcesPlugin
+import de.fhdo.lemma.service.open_api.LemmaGenerator
+
+class SpecifyUrlDialog extends TitleAreaDialog {
+    static val MIN_DIALOG_WIDTH = 500
+    static val MIN_DIALOG_HEIGHT = 250
+    Text txtUrl
+
+    Text txtTargetLocation
+
+    Text txtDataModelName
+
+    Text txtServiceModelName
+
+    Text txtTechnologyModelName
+
+    public URL fetchUrl
+    public String targetLocation
+    public String dataName
+    public String serviceName
+    public String servicePrefix
+    public String technologyName
+
+    Button btnBrowseLocation
+
+    Text txtServicePrefix
+
+    new(Shell parentShell) {
+        super(parentShell)
+    }
+
+    /**
+     * OK button was pressed
+     */
+    override okPressed() {
+        if(saveInput()) {
+             try {
+                val generator = new LemmaGenerator()
+                val parsingMessages = generator.parse(fetchUrl.toString)
+                MessageDialog.openInformation(this.shell, "Parsing Report",
+                    '''«FOR msg : parsingMessages»
+                    «msg»
+                    «ENDFOR»'''
+                )
+                if(generator.isParsed){
+                generator.generateModels('''«targetLocation»/''', '''«dataName».data''',
+                    '''«serviceName».services''','''«technologyName».technology''', servicePrefix)
+                } else {
+                    MessageDialog.openError(this.shell, "Parsing Error",
+                        '''It was not possible to generate an in-memory '''+
+                        '''representation of the file located at «fetchUrl.toString» .'''
+                    )
+                }
+
+            } catch (Exception ex) {
+                MessageDialog.openError(this.shell, "Error",
+                '''An error occured during extraction...«ex.printStackTrace»''')
```

Comment:
Indent. Also: This message will look strange with the three dots. Why not just `during extraction:\n<<...`?